flycheck.el

```diff
@@ -3942,8 +3967,10 @@ Return a list with the contents of the table cell."
                    column 'flycheck-error-list-column-number)
                   (flycheck-error-list-make-cell
                    (symbol-name (flycheck-error-level error)) level-face)
-                  (flycheck-error-list-make-cell
-                   id-str 'flycheck-error-list-id id-str)
+                  ;; Error ID has a button to explain the error
+                  (list id-str
+                        'type 'flycheck-error-list-explain-error
```

Comment:
@lunaryorn Previously it was a button that linked to the error, as are all the cells in the error list really.  They all use the `-error-list` button type that has `-error-list-button-goto-error` as an action.

What I did was add another button type (`-error-list-explain-error`) with the action `-error-list-button-explain-error`, but without changing the face.  So previously the error id linked to the error, and now it linked to the explanation (by calling `-display-error-explanation`).  

Without any explainer present, the button would do nothing, and I thought that was what @cpitclaudel was finding confusing.  So I've then added a different face (`-error-list-id-with-explainer`) to error ids when an explainer is present; by default it looks like a clickable button. 

Here is how it looks when the explainer is present (we *can* actually just drop images and Github will host them; I didn't notice that before):

![screen](https://cloud.githubusercontent.com/assets/340129/19070984/28ee80e0-8a2e-11e6-8aa1-7608db3bb114.png)

It's just a face, so we can tweak it.  Or we can drop it altogether.